tests/gtest/aviftest_helpers.cc

```diff
@@ -230,15 +231,30 @@ bool AreImagesEqual(const avifImage& image1, const avifImage& image2,
       image1.clli.maxPALL != image2.clli.maxPALL) {
     return false;
   }
-  if (image1.transformFlags != image2.transformFlags ||
+  if ((image1.transformFlags & AVIF_TRANSFORM_PASP) !=
+          (image2.transformFlags & AVIF_TRANSFORM_PASP) ||
       ((image1.transformFlags & AVIF_TRANSFORM_PASP) &&
-       std::memcmp(&image1.pasp, &image2.pasp, sizeof(image1.pasp))) ||
+       (image1.pasp.hSpacing != image2.pasp.hSpacing ||
+        image1.pasp.vSpacing != image2.pasp.vSpacing))) {
+    return false;
+  }
+  if ((image1.transformFlags & AVIF_TRANSFORM_CLAP) !=
+          (image2.transformFlags & AVIF_TRANSFORM_CLAP) ||
       ((image1.transformFlags & AVIF_TRANSFORM_CLAP) &&
-       std::memcmp(&image1.clap, &image2.clap, sizeof(image1.clap))) ||
-      ((image1.transformFlags & AVIF_TRANSFORM_IROT) &&
-       std::memcmp(&image1.irot, &image2.irot, sizeof(image1.irot))) ||
-      ((image1.transformFlags & AVIF_TRANSFORM_IMIR) &&
-       std::memcmp(&image1.imir, &image2.imir, sizeof(image1.imir)))) {
+       (image1.clap.widthN != image2.clap.widthN ||
+        image1.clap.widthD != image2.clap.widthD ||
+        image1.clap.heightN != image2.clap.heightN ||
+        image1.clap.heightD != image2.clap.heightD ||
+        image1.clap.horizOffN != image2.clap.horizOffN ||
+        image1.clap.horizOffD != image2.clap.horizOffD ||
+        image1.clap.vertOffN != image2.clap.vertOffN ||
+        image1.clap.vertOffD != image2.clap.vertOffD))) {
+    return false;
+  }
+  // Different rotation and mirroring properties could lead to the same
+  // orientation. Convert to Exif orientation for an easy comparison.
```

Comment:
IMPORTANT: This is also true for the fractions in the `clap` property. If we allow different but equivalent representations, we can call `avifFractionSimplify()` to simplify the fractions before comparing them.

On the other hand, if `AreImagesEqual()` means "the images are identical" rather than "the images are equivalent", then we should require all struct members to be equal. It seems that this is now `AreImagesEqual()` is used by the tests. Please double check.